homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -0,0 +1,237 @@
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_PIN,
+    CONF_WEBFSAPI_URL,
+    DEFAULT_PIN,
+    DEFAULT_PORT,
+    DOMAIN,
+    SSDP_ATTR_SPEAKER_NAME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+STEP_DEVICE_CONFIG_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_PIN,
+            default=DEFAULT_PIN,
+        ): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Frontier Silicon Media Player."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize flow."""
+
+        self._webfsapi_url: str | None = None
+        self._name: str | None = None
+        self._unique_id: str | None = None
+
+    async def async_step_import(self, import_info: dict[str, Any]) -> FlowResult:
+        """Handle the import of legacy configuration.yaml entries."""
+
+        device_url = f"http://{import_info[CONF_HOST]}:{import_info[CONF_PORT]}/device"
+        try:
+            self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+        except FSConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        try:
+            afsapi = AFSAPI(self._webfsapi_url, import_info[CONF_PIN])
+
+            self._unique_id = await afsapi.get_radio_id()
+        except FSConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except InvalidPinException:
+            return self.async_abort(reason="invalid_auth")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        # For manually added devices the unique_id is the radio_id,
+        # for devices discovered through SSDP it is the UDN
+        await self.async_set_unique_id(self._unique_id, raise_on_progress=False)
+        self._abort_if_unique_id_configured()
+
+        self._name = import_info[CONF_NAME] or "Radio"
+
+        _LOGGER.warning("Frontier Silicon %s imported from YAML config", self._name)
+        return await self._create_entry(pin=import_info[CONF_PIN])
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step of manual configuration."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
```

Comment:
Why not do it the other way around? So checking for `if user_input:`. This way you don't need `return self.async_show_form(..)` twice.